src/Sylius/Component/Core/spec/Promotion/Applicator/UnitsPromotionAdjustmentsApplicatorSpec.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\Core\Promotion\Applicator;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Component\Core\Distributor\IntegerDistributorInterface;
+use Sylius\Component\Core\Model\AdjustmentInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\OrderItemUnitInterface;
+use Sylius\Component\Core\Promotion\Applicator\UnitsPromotionAdjustmentsApplicator;
+use Sylius\Component\Core\Promotion\Applicator\UnitsPromotionAdjustmentsApplicatorInterface;
+use Sylius\Component\Order\Factory\AdjustmentFactoryInterface;
+use Sylius\Component\Originator\Originator\OriginatorInterface;
+use Sylius\Component\Promotion\Model\PromotionInterface;
+
+/**
+ * @mixin UnitsPromotionAdjustmentsApplicator
+ *
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class UnitsPromotionAdjustmentsApplicatorSpec extends ObjectBehavior
+{
+    function let(
+        AdjustmentFactoryInterface $adjustmentFactory,
+        IntegerDistributorInterface $distributor,
+        OriginatorInterface $originator
+    ) {
+        $this->beConstructedWith($adjustmentFactory, $distributor, $originator);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Component\Core\Promotion\Applicator\UnitsPromotionAdjustmentsApplicator');
+    }
+
+    function it_implements_units_promotion_adjustments_applicator_interface()
+    {
+        $this->shouldImplement(UnitsPromotionAdjustmentsApplicatorInterface::class);
+    }
+
+    function it_applies_promotion_adjustments_on_all_units_of_given_order(
+        AdjustmentFactoryInterface $adjustmentFactory,
+        AdjustmentInterface $firstAdjustment,
+        AdjustmentInterface $secondAdjustment,
+        AdjustmentInterface $thirdAdjustment,
+        IntegerDistributorInterface $distributor,
+        OrderInterface $order,
+        OrderItemInterface $coltItem,
```

Comment:
Why do you want to have a colt as a order item? Do you want to buy a gun? Guns are dangerous... But not as dangerous as a people who has them. 

> Guns don't kill people...